core/src/main/java/org/apache/iceberg/BaseEntriesTable.java

```diff
@@ -192,37 +188,46 @@ public <T> Boolean notNaN(BoundReference<T> ref) {
 
       @Override
       public <T> Boolean lt(BoundReference<T> ref, Literal<T> lit) {
-        return compareDateFileContent(ref, lit, compareResult -> compareResult < 0);
+        return ROWS_MIGHT_MATCH;
       }
 
       @Override
       public <T> Boolean ltEq(BoundReference<T> ref, Literal<T> lit) {
-        return compareDateFileContent(ref, lit, compareResult -> compareResult <= 0);
+        return ROWS_MIGHT_MATCH;
       }
 
       @Override
       public <T> Boolean gt(BoundReference<T> ref, Literal<T> lit) {
-        return compareDateFileContent(ref, lit, compareResult -> compareResult > 0);
+        return ROWS_MIGHT_MATCH;
       }
 
       @Override
       public <T> Boolean gtEq(BoundReference<T> ref, Literal<T> lit) {
-        return compareDateFileContent(ref, lit, compareResult -> compareResult >= 0);
+        return ROWS_MIGHT_MATCH;
       }
 
       @Override
       public <T> Boolean eq(BoundReference<T> ref, Literal<T> lit) {
-        return compareDateFileContent(ref, lit, compareResult -> compareResult == 0);
+        if (fileContent(ref)) {
+          Literal<Integer> intLit = lit.to(Types.IntegerType.get());
+          Integer fileContentId = intLit.value();
+          if (fileContentId == FileContent.DATA.id()) {
+            return manifestContentId == ManifestContent.DATA.id();
+          } else {
```

Comment:
I know it probably wont happen, but how about we make this code handle future additions.

(ie, add condition  else if with fileContentId == FileContent.DELETE.id(), and then return ROWS_MIGHT_MATCH if neither of these cases. )

Reply:
Sounds good, but I think then we are back to original set `DELETE_CONTENT_SET ` we had or we can inline as suggested, because that we have both equality delete and positional delete in FileContent enum but only one type of delete in manifestContent.